### PR TITLE
Remove reliance on SapphireTest::is_running_test

### PR DESCRIPTION
**Pull request by dhensby**
A first attempt at removing the coupling of SapphireTest with core code.

This should allow us to remove the `PHPUnitShim` file

### CODE REVIEW

**Comment by tractorcow** [comment]
Also https://github.com/silverstripe/silverstripe-framework/issues/6869

File: src/Control/Director.php [hunk 1]
```diff
@@ -1207,6 +1213,25 @@ public static function session_environment()
     }
 
     /**
+     * Determines if unit tests are currently run, flag set during test bootstrap.
```
Comment (sminnee) [inline src/Control/Director.php]: More Director bloat feels like a step in the wrong direction.

**Comment by tractorcow** [comment]
> It probably would logically live in the same place as wherever isDev() / isTest() / isLive() belong, which should also be refactored out of director.

It needs to go into App object doesn't it @sminnee ? What should we do in the mean time; Add it as a future story for once app is in core?

**Comment by sminnee** [comment]
> It needs to go into App object doesn't it @sminnee ? What should we do in the mean time; Add it as a future story for once app is in core?

I guess that would be the logical place — maybe just note that this method should be shifted on the App object card? Or park this work until App object is done?

**Comment by tractorcow** [comment]
I think parking it is best for the sake of velocity.

**Comment by dhensby** [comment]
I wanted to get this in place to be able to remove our PHPUnit shim - if the App object is likely to come soon then we can put this on hold

**Comment by tractorcow** [comment]
It's marked for beta1. https://github.com/silverstripe/silverstripe-framework/issues/6681

Once I finish the injector service upgrade work I'll jump onto it.

**Comment by tractorcow** [comment]
@dhensby might be worth thinking about what other parts of Director should be in app :)

**Comment by dhensby** [comment]
> @dhensby might be worth thinking about what other parts of Director should be in app :)

all of it? :P

**Comment by tractorcow** [comment]
Well, how about anything not routing / url related?

 - url generation, yes?
 - environment state, no
 - routing, yes
 - file paths/handling, no (we need a new director equivalent for file paths IMO).

**Comment by tractorcow** [comment]
This API has been superseded by already merged changes. is_running_tests() has been removed, rather than replaced. :) Any code which relies on configurable behaviour has been moved to be controlled by config, which tests can set on an as-needed basis.

**Pull request closed without merging**